mpop_kiosk/BilingualText.qml

```diff
@@ -10,26 +10,6 @@ import QtQuick 2.0
 Item {
     property string textFr: ""
     property string textEn: ""
-    property string language: "fr"
-    property string text: ""
-
-    Component.onCompleted: {
-        text = language == "fr" ? textFr : textEn;
-    }
-
-    onLanguageChanged: {
-        text = language == "fr" ? textFr : textEn;
-    }
-
-    onTextEnChanged: {
-        if (language == "en") {
-            text = textEn;
-        }
-    }
-
-    onTextFrChanged: {
-        if (language == "fr") {
-            text = textFr;
-        }
-    }
+    property string language: window.lang
```

Comment:
We should avoid accessing properties of a parent item. Instead, let's create a property for this item.

Reply:
A possible alternative would be to use a singleton. In this case, the langue *is* global for the kiosk, so we could very well do that.

Reply:
Though, this should probably be done in another PR.

Reply:
though as far as I know, singleton properties are read-only, so unless I'm mistaken here, maybe this isn't the ideal approach, since `window.lang` IS expected to change, even if not frequently.

from a strictly logical perspective, I would be enclined to keep property bindings as direct as possible instead of creating a local page property alias and then referencing _that_ property in concurring `BilingualText` items. unless there are performance benefits I'm unaware of, I feel like it really only adds an extra step in the evaluation process, and is ultimately more likely to make the code sparse and confusing; and since we _do_ want `BilingualText.qml` to behave as a replacement to the traditional QML translation system, it only seems reasonable to me to have its default language property directly set to the global language setting, and then change its per-instance value wherever (and if ever) needed.

that being said, I would be interested to know if the process of "mediating" global variables does have, as I've mentioned earlier, any sort of observable impact on performance or anything else.

Reply:
Ok: well, if it works, let's just document very well the window.lang property in main.qml, so that we make sure that no one renames it by mistake.